java-checks/src/main/java/org/sonar/java/checks/MethodIdenticalImplementationsCheck.java

```diff
@@ -0,0 +1,156 @@
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (!hasSemantic()) {
+      return;
+    }
+    List<MethodTree> methods = ((ClassTree) tree).members().stream()
+      .filter(member -> member.is(Tree.Kind.METHOD))
+      .map(MethodTree.class::cast)
+      .filter(MethodIdenticalImplementationsCheck::isDuplicateCandidate)
+      .collect(Collectors.toList());
+    if (methods.size() <= 1) {
+      return;
+    }
+    Set<MethodTree> reported = new HashSet<>();
+    for (int i = 0; i < methods.size(); i++) {
+      MethodTree method = methods.get(i);
+      SyntaxToken methodIdentifier = method.simpleName().identifierToken();
+      List<StatementTree> methodBody = method.block().body();
+      methods.stream()
+        .skip(i + 1L)
+        // avoid reporting multiple times
+        .filter(otherMethod -> !reported.contains(otherMethod))
+        // skip overloads
+        .filter(otherMethod -> !methodIdentifier.text().equals(otherMethod.simpleName().name()))
+        .filter(otherMethod -> SyntacticEquivalence.areEquivalent(methodBody, otherMethod.block().body()))
+        .forEach(otherMethod -> {
+          reported.add(otherMethod);
+          reportIssue(
+            otherMethod.simpleName(),
+            String.format(ISSUE_MSG, methodIdentifier.text(), methodIdentifier.line()),
+            Collections.singletonList(new JavaFileScannerContext.Location("original implementation", methodIdentifier)),
+            null);
+        });
+    }
+  }
+
+  private static boolean isDuplicateCandidate(MethodTree method) {
+    BlockTree block = method.block();
+    if (block == null) {
+      return false;
+    }
+    List<StatementTree> statements = block.body();
+    if (statements.isEmpty()) {
+      return false;
+    }
+    if (statements.size() == 1) {
+      StatementTree singleStatement = statements.get(0);
+      return !isTrivialReturn(singleStatement)
+        && !isAnyThrowStatement(singleStatement)
+        && !isMethodInvocationWithoutParameter(singleStatement);
+    }
+    return true;
+  }
+
+  private static boolean isTrivialReturn(Tree tree) {
+    if (!tree.is(Tree.Kind.RETURN_STATEMENT)) {
+      return false;
+    }
+    ExpressionTree returnExpression = ((ReturnStatementTree) tree).expression();
+    return returnExpression == null
+      || returnExpression.is(Tree.Kind.BOOLEAN_LITERAL, Tree.Kind.NULL_LITERAL)
+      || isThis(returnExpression)
+      || isEmptyString(returnExpression)
+      || isZeroOrOne(returnExpression)
+      || isMethodInvocationWithoutParameter(returnExpression);
+  }
+
+  private static boolean isThis(Tree tree) {
+    return tree.is(Tree.Kind.IDENTIFIER) && "this".equals(((IdentifierTree) tree).name());
+  }
+
+  private static boolean isEmptyString(Tree tree) {
+    return tree.is(Tree.Kind.STRING_LITERAL) && "\"\"".equals(((LiteralTree) tree).value());
+  }
+
+  private static boolean isZeroOrOne(Tree tree) {
+    Tree expr = tree;
```

Comment:
I think you should reuse LiteralUtils.longLiteralValue rather than going for a regexp.